src/menu/menu-section.tsx

```diff
@@ -0,0 +1,27 @@
+import { Label, Section } from "./menu-section.styles";
+import { MenuSectionProps } from "./types";
+
+export const MenuSection = ({
+    children,
+    label,
+    showDivider = true,
+    "data-testid": testId = "menu-section",
+    ...otherProps
+}: MenuSectionProps): JSX.Element => {
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    return (
+        <Section
+            $showDivider={showDivider}
+            data-testid={testId}
+            role="group"
+            {...otherProps}
+        >
+            {label && <Label weight="semibold">{label}</Label>}
```

Comment:
something that I missed out: we'll also need to associate the section with the label, so that the screenreader will read out "Menu link, Category 1, group" label when a list item in the section is first focused

this can be done by using `aria-labelledby` to point to the label. we can generate a unique id with:
```
const [internalId] = useState(() => SimpleIdGenerator.generate());
```

the usage would be something like this:
```
<Section aria-labelledby={internalId}>
  <Label id={internalId}>
```

Reply:
updated